memtable/skiplistrep.cc

```diff
@@ -95,6 +95,37 @@ class SkipListRep : public MemTableRep {
     return (end_count >= start_count) ? (end_count - start_count) : 0;
   }
 
+  void UniqueRandomSample(const uint64_t& num_entries,
+                          const uint64_t& sample_size,
+                          std::unordered_set<const char*>* entries) override {
+    entries->clear();
+    SkipListRep::Iterator iter(&skip_list_);
+    // If the sample size is at least 25% of the total number of elements
```

Comment:
(My threshold guesstimate would be more like ~10%.)

Reply:
I am going for 10% - the `sample_size`<<sqrt(`num_entries`) from the reference https://cs.stackexchange.com/a/129757 is not satifying for our situation (if N=1000, then `sample_size` of 2 would redirect to the first sampling algorithm and not the random picks...)